realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -74,7 +74,8 @@ static RealmCacheType valueOf(Class<? extends BaseRealm> clazz) {
 
     // Column indices are cached to speed up opening typed Realm. If a Realm instance is created in one thread, creating
     // Realm instances in other threads doesn't have to initialize the column indices again.
-    private ColumnIndices typedColumnIndices;
+    private static final int MAX_ENTRIES_IN_TYPED_COLUMN_INDICES_ARRAY = 4;
```

Comment:
The reason is `4` is not so big number and no specific reason...
Current implementation uses linear search when finding entry from array or storing entry to array. 

I think 2 or 3 is OK in most cases since schema change does not happen so often while the process of mobile app is living.